### ISSUE TITLE
질문이 있어요

### ISSUE THREAD

**Issue by wseungjin** [taki0112/SphereGAN-Tensorflow#2]
안녕하세요 코드보고 공부하고 있는 학생입니다 
여쭤보고 싶은게 있어서 질문드립니다

```python
""" Training """
        # divide trainable variables into a group for D and a group for G
        t_vars = tf.trainable_variables()
        d_vars = [var for var in t_vars if 'discriminator' in var.name]
        g_vars = [var for var in t_vars if 'generator' in var.name]
```

이부분을 각각의 loss 값을 구해 최소화 하면서 weight값을 옵티마이즈 시켜준다고 이해했고 이 vars 의 값들은 weight값이라고 이해했는데

```python
t_vars = tf.trainable_variables()
for var in t_vars: 
     if 'discriminator' in var.name:
           d_vars = [var]
for var in t_vars: 
     if 'generator' in var.name:
           g_vars = [var]
```

위와 밑의 코드는 같은 코드인가요?
제가 파이썬 문법을 제대로 잘 몰라서 이해가 잘 안되서 
일단 밑처럼 이해하고 변환 했는데 제가 이해하고 있는 내용이 맞는지 
궁금하여 질문드립니다
감사합니다

**Comment by taki0112** [comment]
다음과 같이 해야합니다.
```python
t_vars = tf.trainable_variables()
d_vars = []
g_vars = []
for var in t_vars: 
     if 'discriminator' in var.name:
           d_vars.append(var)
for var in t_vars: 
     if 'generator' in var.name:
           g_vars.append(var)
```

**Comment by wseungjin** [comment]
감사합니다!! 덕분에 잘 공부하고 있습니다
